File: client-scalaz/src/main/scala/org/openapitools/client/api/RetailerProfile.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import RetailerProfile._

case class RetailerProfile (
  detailsHeader: Option[String],
detailsBody: Option[String],
detailsLineLeft1: Option[String],
detailsLineLeft2: Option[String],
detailsLineRight1: Option[String],
metaDescription: Option[String],
directions: Option[String],
hours: Option[String],
homePhone: Option[String],
cellPhone: Option[String],
cellCarrier: Option[CellCarrier],
businessPhone: Option[String],
businessPhoneExt: Option[String],
faxNumber: Option[String],
timeZone: Option[String],
utcOffset: Option[String],
code501c3: Option[String],
emailAddress: Option[String],
address: Option[Address],
web: Option[String],
featured: Option[Boolean],
listed: Option[Boolean])

object RetailerProfile {
  import DateTimeCodecs._

  implicit val RetailerProfileCodecJson: CodecJson[RetailerProfile] = CodecJson.derive[RetailerProfile]
  implicit val RetailerProfileDecoder: EntityDecoder[RetailerProfile] = jsonOf[RetailerProfile]
  implicit val RetailerProfileEncoder: EntityEncoder[RetailerProfile] = jsonEncoderOf[RetailerProfile]
}

File: client-clojure/src/sirqul_io_t_platform/specs/retailer_profile.clj
(ns sirqul-io-t-platform.specs.retailer-profile
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.cell-carrier :refer :all]
            [sirqul-io-t-platform.specs.address :refer :all]
            )
  (:import (java.io File)))


(def retailer-profile-data
  {
   (ds/opt :detailsHeader) string?
   (ds/opt :detailsBody) string?
   (ds/opt :detailsLineLeft1) string?
   (ds/opt :detailsLineLeft2) string?
   (ds/opt :detailsLineRight1) string?
   (ds/opt :metaDescription) string?
   (ds/opt :directions) string?
   (ds/opt :hours) string?
   (ds/opt :homePhone) string?
   (ds/opt :cellPhone) string?
   (ds/opt :cellCarrier) cell-carrier-spec
   (ds/opt :businessPhone) string?
   (ds/opt :businessPhoneExt) string?
   (ds/opt :faxNumber) string?
   (ds/opt :timeZone) string?
   (ds/opt :utcOffset) string?
   (ds/opt :code501c3) string?
   (ds/opt :emailAddress) string?
   (ds/opt :address) address-spec
   (ds/opt :web) string?
   (ds/opt :featured) boolean?
   (ds/opt :listed) boolean?
   })

(def retailer-profile-spec
  (ds/spec
    {:name ::retailer-profile
     :spec retailer-profile-data}))
